src/marqo/core/monitoring/monitoring.py

```diff
@@ -154,6 +154,7 @@ def _get_vespa_health(self, hostname_filter: Optional[str]) -> VespaHealthStatus
         )
 
     def get_cuda_info(self) -> MarqoCudaInfoResponse:
+        # TODO [Refactoring device logic] move this logic to device manager
```

Comment:
It seems like this method overlaps a lot of functionality of `cuda_device_health_check`. But yes, refactoring can be another ticket